### PR TITLE
Core: Remove deprecated AssertHelpers

### PR DESCRIPTION
**Pull request by liuxiaocs7**
- SubTask of https://github.com/apache/iceberg/issues/7094
Remove `AssertHelpers` in `iceberg-core` module.

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestManifestReaderStats.java [hunk 1]
```diff
@@ -275,9 +275,6 @@ private void assertStatsDropped(DataFile dataFile) {
 
   private void assertNullRecordCount(DataFile dataFile) {
     // record count is a primitive type, accessing null record count will throw NPE
-    AssertHelpers.assertThrows(
-        "Should throw NPE when accessing non-populated record count field",
-        NullPointerException.class,
-        dataFile::recordCount);
+    Assertions.assertThatThrownBy(dataFile::recordCount).isInstanceOf(NullPointerException.class);
```
[the remark this answers is not in the document]
Comment (liuxiaocs7) [reply core/src/test/java/org/apache/iceberg/TestManifestReaderStats.java:278]: Thanks for the comment. but there seems no any error message in this case.

[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/TestManifestReaderStats.java:278]: then let's add `.hasMessage(null)` to indicate that

[the remark this answers is not in the document]
Comment (liuxiaocs7) [reply core/src/test/java/org/apache/iceberg/TestManifestReaderStats.java:278]: yep, i use `.hasMessage(null)` in https://github.com/apache/iceberg/pull/7868/commits/ed4c890a4f6225d94f898a2260ba3712706eb821, it works in my local env, but fails in this action https://github.com/apache/iceberg/actions/runs/5322419639/jobs/9638844493

image

any suggestions? thanks!

[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/TestManifestReaderStats.java:278]: it seems that newer JDK versions actually provide an informative error msg to indicate where the NPE happened. In this case let's omit the `.hasMessage()` part

**Review by nastra — APPROVED** [review]
LGTM, thanks @liuxiaocs7

**Pull request merged**